python/lsst/rubintv/production/endOfDay.py

```diff
@@ -0,0 +1,281 @@
+        but must be creatable with write privileges.
+    doRaise `bool`
+        Raise on error?
+    """
+    catchupPeriod = 300  # in seconds, so 5 mins
+    loopSleep = 20
+    endOfDayDelay = 600
+    allSkyDeletionExtraSleep = 1800  # 30 mins
+
+    def __init__(self, *,
+                 allSkyPngRoot=None,
+                 moviePngRoot=None,
+                 dryRunForDeletion=True,
+                 doRaise=False,
+                 **kwargs):
+        self.uploader = Uploader()
+        self.log = _LOG.getChild("backgroundService")
+        self.dryRunForDeletion = dryRunForDeletion
+        self.allSkyPngRoot = allSkyPngRoot
+        self.moviePngRoot = moviePngRoot
+        self.doRaise = doRaise
+        self.butler = butlerUtils.makeDefaultLatissButler()
+        self.bestEffort = BestEffortIsr(**kwargs)
+        self.uploader = Uploader()
+
+    def _raiseIf(self, error):
+        """Raises the error if self.doRaise otherwise logs it as a warning.
+
+        error : ``
+            The error that has been raised.
+
+        Raises
+        ------
+        AnyException
+            Raised if self.doRaise is True, otherwise swallows and warns.
+        """
+        msg = f'Background service error: {error}'
+        if self.doRaise:
+            raise RuntimeError(msg) from error
+        else:
+            self.log.warn(msg)
+
+    def hasDayRolledOver(self):
+        """Check if the dayObs has rolled over.
+
+        Checks if the class' dayObs is the current dayObs and returns False
+        if it is.
+
+        Returns
+        -------
+        hasDayRolledOver : `bool`
+            Whether the day has rolled over.
+        """
+        currentDay = getCurrentDayObs_int()
+        if currentDay == self.dayObs:
+            return False
+        elif currentDay == self.dayObs+1:
+            return True
+        else:
+            raise RuntimeError(f"Encountered non-linear time! Day cleaner was started with {self.dayObs}"
+                               f"and now current dayObs is {currentDay}!")
+
+    def getMissingQuickLookIds(self):
+        """XXX DOCS
+        """
+        allSeqNums = butlerUtils.getSeqNumsForDayObs(self.butler, self.dayObs)
+
+        where = "exposure.day_obs=dayObs AND instrument='LATISS'"
+        expRecords = self.butler.registry.queryDimensionRecords("exposure",
+                                                                where=where,
+                                                                bind={'dayObs': self.dayObs},
+                                                                datasets='quickLookExp')
+        expRecords = list(set(expRecords))
+        foundSeqNums = [r.seq_num for r in expRecords]
+        toMakeSeqNums = [s for s in allSeqNums if s not in foundSeqNums]
+        return [{'day_obs': self.dayObs, 'seq_num': s, 'detector': 0} for s in toMakeSeqNums]
+
+    @staticmethod
+    def _minimizeDataId(dataId):
```

Comment:
This name is confusing to me.  I think part of the reason is that this is doing the opposite of what `daf_butler`'s "expanded" dataIds do (although both "shrink" and "compress" seem like bad verbs too).

Reply:
Yup, agreed, it's awful. I've changed it to `_makeMinimalDataId()` because I think that's most explicit. Let me know if that's still distasteful though.